overlord/snapstate/snapmgr.go

```diff
@@ -62,6 +62,9 @@ type SnapSetup struct {
 	UserID  int       `json:"user-id,omitempty"`
 	Base    string    `json:"base,omitempty"`
 	Type    snap.Type `json:"type,omitempty"`
+	// PlugsOnly indicates whether the relevant revisions for the operation
+	// have only plugs or none, and absolutely no slots.
+	PlugsOnly bool `json:"plugs-only,omitempty"`
```

Comment:
Should we use something like "Flags int" here or "Flags []string" ? But I guess its fine.

Reply:
well we use structs of bools for other flags, don't think it make sense to add more nesting here

Reply:
I'm not sure I parse the comment. Does it mean it has 0 slots and 0 or more plugs?

Reply:
yes

Reply:
could have been called "no-slots",   but my brain seem to think better of "plugs-only" snaps than "no-slots" snaps,  also the former somehow sounds better